File: Healthcare-app/ios/MDBRealmPatient/MDBRealmPatient/Procedure.swift
//
//  Procedure.swift
//  MDBRealmPatient
//
//  Created by Mackbook on 07/05/22.
//

import Foundation
import RealmSwift

class Procedure: Object {
    @Persisted(primaryKey: true) var _id: ObjectId
    @Persisted var encounter: Encounter?
    @Persisted var identifier: String?
    @Persisted var note: List<Procedure_Notes>
    @Persisted var nurseIdentifier: String?
    @Persisted var patientIdentifier: String?
    @Persisted var practitionerIdentifier: String?
    @Persisted var status: String?
    @Persisted var subject: Patient?
    @Persisted var usedReference: List<Medication>
}
class Procedure_Notes: EmbeddedObject {
    @Persisted var author: Reference?
    @Persisted var text: String?
    @Persisted var time: Date?
}
class Medication: EmbeddedObject {
    @Persisted var amount: Int?
    @Persisted var code: Codable_Concept?
    @Persisted var form: Codable_Concept?
}

File: Healthcare-app/android/MDBRealmPatient/app/src/main/java/com/wekanmdb/storeinventory/model/procedure/Procedure.kt
package com.wekanmdb.storeinventory.model.procedure

import com.wekanmdb.storeinventory.model.embededclass.Medication
import com.wekanmdb.storeinventory.model.embededclass.Procedure_Notes
import com.wekanmdb.storeinventory.model.encounter.Encounter
import com.wekanmdb.storeinventory.model.patient.Patient
import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.Required
import org.bson.types.ObjectId

open class Procedure (
    @PrimaryKey var _id: ObjectId = ObjectId(),
    var identifier: String? = null,
    var status: String? = null,
    var subject: Patient? = null,
    var encounter: Encounter? = null,
    var usedReference: RealmList<Medication> = RealmList(),
    var note: RealmList<Procedure_Notes> = RealmList(),
    var patientIdentifier: String? = null,
    var practitionerIdentifier: String? = null,
    var nurseIdentifier: String? = null
): RealmObject() {
}